src/meteodatalab/operators/vertical_extrapolation.py

```diff
@@ -0,0 +1,114 @@
+"""Vertical extrapolation operators."""
+
+# Third-party
+import numpy as np
+import xarray as xr
+
+# Local
+from .. import physical_constants as pc
+
+LAPSE_RATE = 0.0065  # K m^-1
+H1 = 2000.0
+H2 = 2500.0
+T1 = 298.0
+
+
+def extrapolate_temperature_sfc2p(
+    t_sfc: xr.DataArray,
+    z_sfc: xr.DataArray,
+    p_sfc: xr.DataArray,
+    p_target: float,
+) -> xr.DataArray:
+    """Extrapolate temperature to a target pressure level.
+
+    Implements the algorithm described in [1]_. The algorithm extrapolates
+    temperature from the surface to a target pressure level using a
+    polynomial expression of a dimensionless variable y, which is a function of
+    the surface temperature, surface pressure, and geopotential. It assumes
+    a constant lapse rate of 0.0065 K m^-1.
+
+    Parameters
+    ----------
+    t_sfc : xr.DataArray
+        Surface temperature [K].
+    z_sfc : xr.DataArray
+        Surface geopotential.
```

Comment:
Please replace the geopotential with the height. There are other operators that use the surface height rather than the geopotential and it seems that the value is converted to height to compare against the thresholds. Also, please add the unit to every parameter